### PR TITLE
Doc: added service checklist section

### PR DESCRIPTION
**Pull request by jmoreira-valory**
## Proposed changes

Doc: added service checklist section (service readiness).

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: docs/deployment/on-chain_deployment_checklist.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+      benchmark_tool:
+        args:
+          log_dir: ${SKILL_<YOUR_SKILL_NAME>_MODELS_BENCHMARK_TOOL_ARGS_LOG_DIR:str:/benchmarks}
+
+      params:
+        args:
+          consensus:
+            max_participants: ${SKILL_<YOUR_SKILL_NAME>_MODELS_PARAMS_ARGS_CONSENSUS_MAX_PARTICIPANTS:int:4}
+          setup:
+            all_participants: ${SKILL_<YOUR_SKILL_NAME>_MODELS_PARAMS_ARGS_SETUP_ALL_PARTICIPANTS:list:[]}
```
Comment (angrybayblade) [inline docs/deployment/on-chain_deployment_checklist.md:143]: These long environment variable names are not required anymore they're generated automatically below the service level. So these environment variable names needs to be defined at the service level, below the service level they're generated automatically so you just need to define the env placeholder with default value and data type like

```suggestion
            all_participants: ${list:[]}      
```

Comment (jmoreira-valory) [reply docs/deployment/on-chain_deployment_checklist.md:143]: Ok. This only applies to aea-config.yaml ?

Comment (dvilelaf) [reply docs/deployment/on-chain_deployment_checklist.md:143]: I think @angrybayblade is working on a config validation command. Keep that in mind as once it is available we should add a reference to it here.

Comment (angrybayblade) [reply docs/deployment/on-chain_deployment_checklist.md]: @dvilelaf I'll open a ticket once I'm done with the features

@jmoreira-valory Yes

**Review by dvilelaf — DISMISSED** [review]
LGTM, but let's get more reviews.

File: docs/deployment/on-chain_deployment_checklist.md [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+      params:
+        args:
+          consensus:
+            max_participants: ${int:4}
+          setup:
+            all_participants: ${list:[]}      
+            safe_contract_address: ${list:[]}
+          tendermint_url: ${TENDERMINT_URL:str:http://localhost:26657}
+          tendermint_com_url: ${TENDERMINT_COM_URL:str:http://localhost:8080}
+          service_registry_address: ${str:none}
```
Comment (0xArdi) [inline docs/deployment/on-chain_deployment_checklist.md:151]: should this be `none` or `null`?

**Review by angrybayblade — DISMISSED** [review]

**Review by Adamantios — CHANGES_REQUESTED** [review]
Looks good. Let's address @0xArdi 's comment for both `service_registry_address` and `on_chain_service_id`.

**Review by Adamantios — APPROVED** [review]

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**